packages/lib/src/Button.wc.svelte

```diff
@@ -0,0 +1,83 @@
+<svelte:options customElement="sesamy-button" />
+
+<script>
+  export let outline = true;
+  export let loading = false;
+  export let disabled = false;
+  export let onclick = () => {};
+  export let href = "";
+</script>
+
+{#if href && !disabled && !loading}
+  <a {href} class="button" class:outline {onclick}><slot /></a>
+{:else}
+  <button class="button" class:outline class:disabled {disabled} {onclick}>
+    {#if loading}
+      <span class="loader"></span>
+    {/if}
+    <slot />
+  </button>
```

Comment:
It's a little weird to fall back to rendering a button based on states. Arguably:
a) Either both a and button can be disabled and have a loading state
or
b) Just do:
```
{#if href}
 <a {href} class="button" class:outline {onclick}><slot /></a>
{:else}
  <button ... //everything that has to do with button
{/if}

Having the "#if href && !disabled && !loading" is just verbose and makes it more confusing to read
```

Reply:
Interesting, so this is part AI magic, but I'm thinking that this could be a a11y-pattern? I did see the same in for instance the flowbite button component: https://github.com/themesberg/flowbite-svelte/blob/main/src/lib/buttons/Button.svelte

So, I'm thinking: If it has a href it should really be a a-tag so that screen readers and other stuff reads it correctly and you for instance can do "open in new tab".  But.. if it's disabled or in a loading state you don't really want that so in that case we render it as a disabled button instead. I'm guessing there is no such thing as a disabled a-tag?